### ISSUE TITLE
How to evaluate the result of dataset that does not have any labels

### ISSUE THREAD

**Issue by Yarmohamadshr** [UKPLab/sentence-transformers#2359]
Hi,

I was trying to look at the different evaluation metrics that are provided to SentenceTransformers. I have a column of text in my dataset that I compare against a query and get the top k similarity using cosine similarity. I do not know if there is any method to evaluate the result. Should I consider the cosine similarity score as my evaluation metric as well? By evaluation, I mean, how can I show that the result I got is good? Is reasonable?

from sentence_transformers import SentenceTransformer, util
import pandas as pd

# Load a pre-trained model
model = SentenceTransformer('msmarco-distilbert-cos-v5')

# Example query
query = "Semantic search example query"

# Example corpus
corpus = ["Example sentence 1", "Example sentence 2", "Example sentence 3", ...]  # Add more sentences to your corpus

# Encode the query and corpus into embeddings
query_embedding = model.encode(query, convert_to_tensor=True)
corpus_embeddings = model.encode(corpus, convert_to_tensor=True)

# Compute cosine similarities
cosine_similarities = util.pytorch_cos_sim(query_embedding, corpus_embeddings)[0]

# Get indices of the 3 nearest neighbors
indices_nearest_neighbors = pd.Series(cosine_similarities).nlargest(3).index

# Retrieve the 3 nearest neighbors
nearest_neighbors = [corpus[i] for i in indices_nearest_neighbors]

# Print the results
print(f"Query: {query}")
print("3 Nearest Neighbors:")
for neighbor in nearest_neighbors:
    print("-", neighbor)
